configure.ac

```diff
@@ -901,7 +901,6 @@ AS_IF([test "$GCC" = yes], [
     ], [
 	RUBY_TRY_LDFLAGS([-Wl,-unexported_symbol,_Init_*], [visibility_option=ld], [visibility_option=no])
     ])
-    test "$visibility_option" = no || OBJCOPY=:
```

Comment:
This makes it so that we run OBJCOPY on non-YJIT --enable-shared builds, even when `visibility_option != no`:

https://github.com/ruby/ruby/blob/8872ebec6a3edc8327b5f348656c73a52bceda4a/template/Makefile.in#L339-L344

It should be fine, assuming  OBJCOPY that was found does not corrupt the file (it's a ignore-error command). Maybe we could just remove that OBJCOPY call now that we've [dropped support for GCC 3](https://bugs.ruby-lang.org/issues/18839). GCC 4.0 and up take has `-fvisibility`. Not sure when Clang started to support the option, though.

Reply:
This fallback is needed when `objcopy` is not installed.

```diff
diff --git a/configure.ac b/configure.ac
index addb98c596a..2eac961793d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -254,7 +254,7 @@ AC_CHECK_TOOLS([AR],      [gar ar])
 AC_CHECK_TOOLS([AS],      [gas as])
 AC_CHECK_TOOLS([LD],      [gld ld]) # ... try gold ?
 AC_CHECK_TOOLS([NM],      [gnm nm])
-AC_CHECK_TOOLS([OBJCOPY], [gobjcopy objcopy])
+AC_CHECK_TOOLS([OBJCOPY], [gobjcopy objcopy], [:])
 AC_CHECK_TOOLS([OBJDUMP], [gobjdump objdump])
 AC_CHECK_TOOLS([STRIP],   [gstrip strip], [:])
 
```